app/src/androidTest/java/com/duckduckgo/app/autocomplete/api/AutoCompleteApiTest.kt

```diff
@@ -48,31 +45,94 @@ class AutoCompleteApiTest {
+    fun whenAutoCompleteReturnsMultipleBookmarkHitsThenLimitToMaxOfTwo() {
+        whenever(mockAutoCompleteService.autoComplete("title")).thenReturn(Observable.just(emptyList()))
+        whenever(mockBookmarksDao.bookmarksObservable()).thenReturn(
+            Single.just(
+                listOf(
+                    BookmarkEntity(0, "title", "https://example.com"),
+                    BookmarkEntity(0, "title", "https://foo.com"),
+                    BookmarkEntity(0, "title", "https://bar.com"),
+                    BookmarkEntity(0, "title", "https://baz.com")
+                )
+            )
+        )
+
+        val result = testee.autoComplete("title").test()
+        val value = result.values()[0] as AutoCompleteResult
+
+        assertEquals(
+            listOf(
+                AutoComplete.AutoCompleteSuggestion.AutoCompleteBookmarkSuggestion(phrase = "example.com", "title", "https://example.com"),
+                AutoComplete.AutoCompleteSuggestion.AutoCompleteBookmarkSuggestion(phrase = "foo.com", "title", "https://foo.com"),
+            ),
+            value.suggestions
+        )
+    }
+
+    @Test
+    fun whenAutoCompleteReturnsDuplicatedItemsThenDedup() {
+        whenever(mockAutoCompleteService.autoComplete("title")).thenReturn(
+            Observable.just(
+                listOf(
+                    AutoCompleteServiceRawResult("example.com"),
+                    AutoCompleteServiceRawResult("foo.com"),
+                    AutoCompleteServiceRawResult("bar.com"),
+                    AutoCompleteServiceRawResult("baz.com")
+                )
+            )
+        )
+        whenever(mockBookmarksDao.bookmarksObservable()).thenReturn(
+            Single.just(
+                listOf(
+                    BookmarkEntity(0, "title example", "https://example.com"),
+                    BookmarkEntity(0, "title foo", "https://foo.com"),
+                    BookmarkEntity(0, "title bar", "https://bar.com")
+                )
+            )
+        )
 
-        val result = testee.autoComplete("foo").test()
+        val result = testee.autoComplete("title").test()
         val value = result.values()[0] as AutoCompleteResult
 
-        assertSame("https://example.com", value.suggestions[0].phrase)
+        assertEquals(
+            listOf(
+                AutoComplete.AutoCompleteSuggestion.AutoCompleteBookmarkSuggestion(phrase = "example.com", "title example", "https://example.com"),
+                AutoComplete.AutoCompleteSuggestion.AutoCompleteBookmarkSuggestion(phrase = "foo.com", "title foo", "https://foo.com"),
+                AutoComplete.AutoCompleteSuggestion.AutoCompleteSearchSuggestion(phrase = "bar.com", true),
+                AutoComplete.AutoCompleteSuggestion.AutoCompleteSearchSuggestion(phrase = "baz.com", true)
+            ),
+            value.suggestions
+        )
```

Comment:
Consider testing the ranking logic in `AutoCompleteApi`. It would be nice to have tests to know how the different scores affect the order in the bookmark results.